### PR TITLE
Update Makefile.common

### PR DESCRIPTION
**Pull request by simonpasquier**
This change also uses the latest staticcheck version which comes with new verifications, hence some clean up in the code.

### CODE REVIEW

File: chunkenc/bstream.go [hunk 1]
```diff
@@ -39,6 +39,8 @@
 // OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+//lint:file-ignore U1000 Ignore all unused code.
```
Comment (krasi-georgiev) [inline chunkenc/bstream.go]: why is this needed?

File: encoding_helpers.go [hunk 2]
```diff
@@ -11,6 +11,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//lint:file-ignore U1000 Ignore all unused code.
```
Comment (krasi-georgiev) [inline encoding_helpers.go]: why is this needed?

File: index/encoding_helpers.go [hunk 3]
```diff
@@ -11,6 +11,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//lint:file-ignore U1000 Ignore all unused code.
```
Comment (krasi-georgiev) [inline index/encoding_helpers.go]: why is this needed?

File: chunkenc/chunk_test.go [hunk 4]
```diff
@@ -32,7 +32,7 @@ func TestChunk(t *testing.T) {
 	for enc, nc := range map[Encoding]func() Chunk{
 		EncXOR: func() Chunk { return NewXORChunk() },
 	} {
-		t.Run(fmt.Sprintf("%s", enc), func(t *testing.T) {
+		t.Run(fmt.Sprintf("%d", enc), func(t *testing.T) {
```
Comment (krasi-georgiev) [inline chunkenc/chunk_test.go]: is there any negative effect if we have this as `t.Run(fmt.Sprintf("%v", enc), func(t *testing.T) {` .
I guess where performance is important it does matter, but in all other cases I prefer to set it as `%v` and forget about it. Less likely to break if the param changes its type.

File: Makefile.common [hunk 5]
```diff
@@ -125,8 +140,12 @@ common-vet:
 .PHONY: common-staticcheck
 common-staticcheck: $(STATICCHECK)
 	@echo ">> running staticcheck"
+	chmod +x $(STATICCHECK)
 ifdef GO111MODULE
-	GO111MODULE=$(GO111MODULE) $(STATICCHECK) -ignore "$(STATICCHECK_IGNORE)" -checks "SA*" $(pkgs)
+# 'go list' needs to be executed before staticcheck to prepopulate the modules cache.
+# Otherwise staticcheck might fail randomly for some reason not yet explained.
```
Comment (krasi-georgiev) [inline Makefile.common]: did you open an issue somewhere to report this weird behaviour?

Comment (simonpasquier) [reply Makefile.common]: Not yet. Still investigating.

File: chunkenc/bstream.go (hunk 1 above)
Comment (simonpasquier) [reply chunkenc/bstream.go]: This file has a few functions that aren't used anywhere but I wasn't sure if we wanted to remove them or not.

```
chunkenc/bstream.go:56:6: func newBWriter is unused (U1000)
chunkenc/bstream.go:60:19: func (*bstream).clone is unused (U1000)
```

File: encoding_helpers.go (hunk 2 above)
Comment (simonpasquier) [reply encoding_helpers.go]:
```
encoding_helpers.go:35:18: func (*encbuf).len is unused (U1000)
encoding_helpers.go:38:18: func (*encbuf).putBytes is unused (U1000)
encoding_helpers.go:41:18: func (*encbuf).putBE32int is unused (U1000)
encoding_helpers.go:42:18: func (*encbuf).putBE64int is unused (U1000)
encoding_helpers.go:44:18: func (*encbuf).putUvarint32 is unused (U1000)
encoding_helpers.go:75:18: func (*encbuf).putHash is unused (U1000)
encoding_helpers.go:94:18: func (*decbuf).uvarint32 is unused (U1000)
encoding_helpers.go:95:18: func (*decbuf).be32int is unused (U1000)
encoding_helpers.go:99:18: func (*decbuf).crc32 is unused (U1000)
encoding_helpers.go:182:18: func (*decbuf).decbuf is unused (U1000)
```

File: index/encoding_helpers.go (hunk 3 above)
Comment (simonpasquier) [reply index/encoding_helpers.go]:
```
index/encoding_helpers.go:34:18: func (*encbuf).putBytes is unused (U1000)
index/encoding_helpers.go:38:18: func (*encbuf).putBE64int is unused (U1000)
index/encoding_helpers.go:39:18: func (*encbuf).putBE64int64 is unused (U1000)
index/encoding_helpers.go:90:18: func (*decbuf).uvarint32 is unused (U1000)
index/encoding_helpers.go:92:18: func (*decbuf).be64int64 is unused (U1000)
index/encoding_helpers.go:165:18: func (*decbuf).byte is unused (U1000)
index/encoding_helpers.go:178:18: func (*decbuf).decbuf is unused (U1000)
```

**Comment by krasi-georgiev** [comment]
IIUC `//lint:file-ignore U1000 Ignore all unused code.` is to bypass some false positives? 
maybe worth opening an issue so they can fix it. The same for the 'go list' workaround.

LGTM

**Comment by simonpasquier** [comment]
> IIUC //lint:file-ignore U1000 Ignore all unused code. is to bypass some false positives?

No those are true warnings. For instance, this function is used nowhere:

https://github.com/prometheus/tsdb/blob/8d991bdc1e494074ca2d559e5a75f49f13a789ac/chunkenc/bstream.go#L56-L58

**Comment by krasi-georgiev** [comment]
so why not remove these?

**Comment by krasi-georgiev** [comment]
@gouthamve 	do you see any problem with removing these unused unexported functions?

File: index/index.go [hunk 6]
```diff
@@ -36,7 +36,6 @@ const (
 	// MagicIndex 4 bytes at the head of an index file.
 	MagicIndex = 0xBAAAD700
 
-	indexFormatV1 = 1
```
Comment (krasi-georgiev) [inline index/index.go]: btw this shouldn't be removed , but refactored like in

https://github.com/prometheus/tsdb/pull/492/files#diff-71ebe2bcf31a915b1fa3b3b289d5d31dR651

**Comment by krasi-georgiev** [comment]
ping @gouthamve

**Comment by krasi-georgiev** [comment]
In that case @simonpasquier would you mind resolving the conflicts and will merge

**Comment by simonpasquier** [comment]
done

**Comment by simonpasquier** [comment]
Grrr, Travis CI is messing around with `GO111MODULE`, I'll look at it.

**Comment by simonpasquier** [comment]
Do not merge before https://github.com/prometheus/prometheus/pull/5191 gets merged.
